numpy/linalg/tests/test_linalg.py

```diff
@@ -923,6 +922,40 @@ def test_future_rcond(self):
             # Warning should be raised exactly once (first command)
             assert_(len(w) == 1)
 
+    @pytest.mark.parametrize(["m", "n", "n_rhs"], [
+        (0, 4, 1),
+        (0, 4, 2),
+        (4, 0, 1),
+        (4, 0, 2),
+        (4, 2, 0),
+        (0, 0, 0)
+    ])
+    def test_empty_a_b(self, m, n, n_rhs):
+        a = np.arange(m * n).reshape(m, n)
+        b = np.ones((m, n_rhs))
+        x, residuals, rank, s = linalg.lstsq(a, b, rcond=None)
+        assert_equal(x.shape, (n, n_rhs))
+        assert_equal(residuals.shape, ((n_rhs,) if m > n else (0,)))
+        if m > n and n_rhs > 0:
+            # residuals are exactly the squared norms of b's columns
+            assert_equal(residuals, np.full(residuals.shape, m))
```

Comment:
Why is the squared norm of each column equal to `m`?

Reply:
When `a` has no columns, the residuals will be exactly `b`.

Reply:
What you have here means that `residuals[i] == m` - is that really sensible?

Reply:
So I'm very surprised this test passes at all

Reply:
The residuals (as normally understood) are the error vectors, so with a column of `m` `1`'s, it works out to `m`. The output "resids" are the squared norms of the error vectors. So that's not surprising.

Reply:
Ah, I missed that `b` is all 1s. I'd prefer this test to express things in terms of `(b*b).sum(axis=-1)` rather than `np.full(residuals.shape, m)` - which also makes it match the comment